### ISSUE TITLE
Mass Dispell  thinking totems as players

### ISSUE THREAD

**Issue by pucko91** [FeenixServerProject/Archangel_2.4.3_Bugtracker#3914]
Report ID: Titum005

Reporter Name: Titum

Date/Time: 161128 11:06ST

Error Severity: Minor until put into context, than it's major.

Zone/Instance: worldwide/sunwell plateau.

What type of error occurred?: Whenever Mass Dispell is cast on a big group of people and if there are totems on the area mass dispell is casts the totems "soak" one of the maximum 10 dispells within that area the totem gets counted as players.

What was the actual result vs. the expected result?: Result is several mass dispells would need to be cast instead of 1 to make sure they geteveryone whom needs dispelling. Exspected result is the mass dispell ignoring totems as players and only dispells players within the circle.

Can the error be repeated? If so, what are the steps?:example 1. Zul'jin when he hits 80% he does a debuff on everyone what will lead to a stun if not dispelled. if there are totems nearby in the dispell area they can count as players and many mass dispells would have to be casts in order to get the debuff away from everyone. Players have learned this and place their totems away from others in order to make mass dispell effective again.

example 2. (hypothetical) Felmyst. Felmyst has an ability called Gas Nova (spell=45855) this is dispellable and since it hits everyone within the bosses room mass dispell is needed to get rid of it because it kills fast and with totems counted as players mass dispell would need to be casted twice or even more causing player deaths on the encounter. Now the only reason I could know that this scenario would happen is because mass dispell targets totems as players already and then it will break one of the bosses mechanics making him either unkillable or adds extra non blizzlike raiding abilities and tactics that players would ahve to work around.

Link to Video Evidence: do not have any but is testable through mass dispelling totems.

External/Related Links: See answer above.

Please detail your character’s race, class, and items (optional): Orc Shaman, Totems and mass dispell.

**Comment by lajhar** [comment]
If I'm correct Mass Dispel is not thinking, it dispels every friendly target (up to 10 counted from the middle of the area), works as intended.

Regarding to your report I think it's another issue, where **Totems should not recieve** (they should be immune to) **any harmful AoE effects**, like Creeping Paralisis (to stay at your example), therfore Mass Dispel would affect only players and pets.

Need confirmation!

**Comment by Chuchote0321** [comment]
https://www.youtube.com/watch?v=6GqvHujTuEs at 1min 55 s 
2 water totems and 2 earth totems receive the mass dispell effet, there are 4 players still have Paralysis debuff after mass dispel.

**Comment by pucko91** [comment]
ty chuck.

**Comment by lanadelray** [comment]
Much like on Illidari council - Veras Darkshadow constantly thinks a totem is a person! Lol

**Comment by Qingle** [comment]
Might be something to do with the TYPEID system and how it interacts with spell flags.
If that helps in any way.

Any source on whether this is what it was like on retail would be grand (I can't personally remember)

**Comment by nessz** [comment]
I could only find this video of SK Gaming vs Felmyst. https://youtu.be/aKi-Dx7cm1w?t=301

Starting of from the timestamp, you can see that Gas Nova was cast and the guys on the screen have it. You can also see a Priest casting Mass Dispell -- all friendly players in range of Mass Dispell get their debuff removed correctly without the totems which are placed there interfering! (play/stop for a few seconds to get a better grasp of the situation)

For reference on the Felmyst encounter:
> Gas Nova - Area-of-Effect spell dealing 1885 to 2115 Nature damage and draining 3000 health and 1000 mana every 2 seconds for 30 secs to everyone in a 50 yard diameter. Dispellable. Should be removed with Priests' Mass Dispel. 

(ref 
http://wowwiki.wikia.com/wiki/Felmyst 
http://wow-one.com/database/index.php?spell=45855)

The current situation on Feenix is, as already stated above and shown in the linked video by @Chuchote0321, that Mass Dispell can and will target totems. This is a major issue in encounters where it is essential to dispell multiple debuffs as quickly as possible. 
It is as if the totems are recognised as players by the targeting system.

**Comment by nessz** [comment]
@lajhar I just tested a few things and it seems like your guess was partially right. From what I could tell totems are immune to AoE damage and Curses. Totems are not immune against any magic based attacks like Hammer of Justice or Creeping Paralysis -- the dispellable kind.
Note the the difference between Curses and Magic based debuffs. 
Also note that I used 0 damage dealing single target spells to test this. I don't have it in my videos (forgot about it), but everyone knows you can kill totems with single target spells. E.g. in PvP ppl kill totems with Ice Lance or Death Coil. Note that you can not apply any periodic damage dealing spells like Corruption to totems. Anyways, therefore I used the 0 damage single target spells for testing.

Totems vs WL: https://youtu.be/8LKcvJQDx48
You can see that totems are immune to AoE damage. They are also immune to Curse based single target spells. I forgot to record this, but totems are immune to Curse of Agony as well (deals periodic dmg and it is a Curse).
Totems vs Paladin: https://youtu.be/WQICpGOzSX8
You can see that the totem is hit by Hammer of Justice (NOT displaying any immune indicator!). I mean it doesn't affect the totem, since it is not a player, but the spells is still applied as it looks like.

I can only guess, but since totems are immune to Curses (damage dealing or not) this seems somehow related to the fact that you can kill totems with any single target **magic based spell**. It looks as if there is no distinction between damaging and non-damaging magic based spells. Therefore totems can be **targeted by any harmful magic based spell**! Basically only damaging single target magic spells should affect totems -- they should just kill it. Any other harmful non-damaging magic spell should not affect totems -- they should be immune to it, e.g. be immune to Hammer of Justice. 
One more thing to how you can think about this. I don't know how it was implemented on retail, but there are two possibilities how the dispell issue can be resolved. One I just described above (make totems immune to non-damaging magic based spells), the other would be to look at how targeting works for Mass Dispell. That means it should not matter if the totem has some magic debuff applied to it internally -- Mass Dispell simply should not see totems as valid dispell targets.

**Comment by lajhar** [comment]
The issue is not limited to damage/debuff magic effects.
Another interesting fact from live that one can buff totems.
For example **you can cast Fortitude on a friendly totem** (practically it has no effect, but the aura gets applied on the totem), which is not blizzlike (I'm 90% sure, but correct me if I'm wrong).  **Therefore** casting **Mass Dispel** by the priest from the opposite faction **will affect the totem** aswell.
That's why I think Mass Dispel is not the bugged part, it wouldn't affect totems, if totems wouldn't recieve magical auras as intended.

**Comment by nessz** [comment]
There you go, a video where I did exactly that. As expected I can buff totems and dispell the magic effect. This time around with combat log. https://youtu.be/wsePq8LJDVE

Of course what you are saying is true, @lajhar. Totems shouldn't be affected by magical auras! Let me rephrase what I meant by my earlier post. 
The expected and blizzlike behaviour should be that **totems should not receive magical auras**. That being said, it might be _more difficult to fix this issue in a blizzlike manner_ than applying a _temporary band-aid fix_ which simply tells Mass Dispel to disregard totems as valid targets. The outcome would be the same and it might be worth thinking about -- in my opinion this still is a **major issue**.

**Comment by Zerixx** [comment]
On my local - as on PTR. I am not seeing this happening.

http://imgur.com/a/m9kI8

**Comment by Zerixx** [comment]
> The expected and blizzlike behaviour should be that totems should not receive magical auras. That being said, it might be more difficult to fix this issue in a blizzlike manner than applying a temporary band-aid fix which simply tells Mass Dispel to disregard totems as valid targets. The outcome would be the same and it might be worth thinking about -- in my opinion this still is a major issue.

Yeah.. you're right. We need to fix this.

**Comment by nessz** [comment]
That is very very interesting @Zerixx, in a strange kind of way. I made my video on the live server minutes before I posted my reply. This needs some further investigation if some change influencing this was pushed onto the PTR.

**Comment by Zerixx** [comment]
Well, if you haven't been aware of my TBC changes recently, I completely rewrote the spell & aura system so it may have happened by circumstance there. @nessz

**Comment by nessz** [comment]
I wasn't! Sounds very promising! 
Looks like Mass Dispell targeting got "smarter".

**Comment by Nifire** [comment]
It Is a genreal issue on the 2.4.3 that pets & totems are targeted by boss spells. If they were not targeted by spells they would not eat the mass dispell either, or am i missing somehing?
